cloudflare/schema_cloudflare_zone_cache_variants.go

```diff
@@ -0,0 +1,42 @@
+package cloudflare
+
+import (
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+var supportedCacheVariantsExtensions = []string{
+	"avif",
+	"bmp",
+	"gif",
+	"jpeg",
+	"jpg",
+	"jpg2",
+	"jp2",
+	"png",
+	"tiff",
+	"tif",
+	"webp",
+}
+
+func resourceCloudflareZoneCacheVariantsSchema() map[string]*schema.Schema {
+	variantsSchema := map[string]*schema.Schema{
+		"zone_id": {
+			Type:     schema.TypeString,
+			Required: true,
+			ForceNew: true,
+		},
+	}
+
+	for _, ext := range supportedCacheVariantsExtensions {
```

Comment:
not really sure how i feel about the dynamic nature of creating schema items due to making it difficult for discovery. i'll have a think about this.

Reply:
:+1: My main goal was not to repeat the whole schema over and over again. An alternative would be to do something like:

```
"avif": extensionSchema(),
"gif": extensionSchema(),
```

I think an upside with this alternative is that the schema is a little bit easier to read. I'll do that, LMK what you think